f8a_report/v2/report_generator.py

```diff
@@ -0,0 +1,488 @@
+        """Parser for worker data for Stack Analyses v2.
+
+        :arg:
+            stacks_data: Stacks Collected from DB within time-frame.
+            frequency: Frequency of Report ( daily/monthly )
+        :return: Final Venus Report Generated.
+        """
+        logger.info("Normalising v2 Stack Data.")
+        stacks_data = json.loads(stacks_data)
+        report_name = self.get_report_name(frequency, self.end_date)
+        report_template = self.get_report_template(self.start_date, self.end_date)
+
+        report_content = self.analyse_stack(stacks_data, report_template)
+
+        self.unique_stacks_with_recurrence_count = {
+            'npm': self.populate_key_count(self.stacks_list['npm']),
+            'maven': self.populate_key_count(self.stacks_list['maven']),
+            'pypi': self.populate_key_count(self.stacks_list['pypi'])
+        }
+        self.unique_stacks_with_deps_count = \
+            self.set_unique_stack_deps_count(self.unique_stacks_with_recurrence_count)
+        self.set_average_response_time()
+
+        # Get a list of unknown licenses
+        for lic_dict in self.flatten_list(self.all_unknown_lic):
+            if 'license' in lic_dict:
+                self.unknown_licenses.append(lic_dict['license'])
+
+        unknown_deps_ingestion_report = UnknownDepsReportHelper().get_current_ingestion_status()
+
+        report_content['stacks_summary'] = self.build_report_summary(
+            unknown_deps_ingestion_report, report_content)
+
+        if frequency == 'monthly':
+            # monthly data collection on the 1st of every month
+            self.collate_raw_data(self.unique_stacks_with_recurrence_count, frequency)
+
+        if retrain:
+            return self.unique_stacks_with_recurrence_count
+
+        venus_input = [frequency, report_name, report_content]
+        logger.info("Venus Report Successfully Generated.")
+        return venus_input
+
+    def get_report(self, start_date, end_date, frequency='daily', retrain=False):
+        """Generate the stacks report: Worker Report and Ingestion Report.
+
+        :param start_date: Date from where to start collecting stacks.
+        :param end_date: Date upto which to collect stacks.
+        :param frequency: Frequency of Reporting (daily/ monthly)
+        :param retrain: Boolean to retrain Model.
+        :returns: Worker Results and Ingestion Results
+        """
+        logger.info("Get Report Executed.")
+        self.start_date = start_date
+        self.end_date = end_date
+        rds_obj = ReportQueries()
+        s3_obj = S3Helper()
+        ids = rds_obj.retrieve_stack_analyses_ids(start_date, end_date)
+        worker = 'stack_aggregator_v2'
+        ingestion_results = False
+        ingestion_data = {}
+
+        if frequency == 'daily':
+            start = dt.now()
+
+            # Ingestion Reporting
+            ingestion_data['EPV_DATA'] = rds_obj.retrieve_ingestion_results(start_date, end_date)
```

Comment:
We dont need the ingestion report part in this file as it has not changed. This already exists in the old file and can be reused as it is.

Reply:
We have a change in Ingestion Reporting too. 
Decoupled few functions:
- decoupled queries function out in v2, 
- decoupled s3 saving function out in v2. 

All is done in view of Sunsetting existing scenario(v1) soon.

Reply:
I guess only the saving part has changed. We ideally need 2 files , one for processing the ingestion report and one for the stacks report.

Reply:
Ack! Removing Ingestion functionality completely from v2.